lib/ansible/modules/crypto/openssl_certificate.py

```diff
@@ -830,11 +831,14 @@ def _validate_issuer():
                     )
 
         def _validate_has_expired():
-            if self.has_expired:
-                if self.has_expired != self.cert.has_expired():
-                    self.message.append(
-                        'Certificate expiration check failed (certificate expiration is %s, expected %s)' % (self.cert.has_expired(), self.has_expired)
-                    )
+            time_string = to_native(self.cert.get_notAfter())
```

Comment:
Maybe add a comment here that this is similar to the PyOpenSSL code for `cert.has_expired()`, but that older versions have a buggy implementation and we thus do it manually here.

Reply:
Added comments, hopefully makes sense :)